packages/core/src/app/pg-core/db/chain-db.ts

```diff
@@ -1,24 +1,523 @@
+/* External Imports */
+import BigNum = require('bn.js')
+
 /* Internal Imports */
-import { KeyValueStore, ChainDB } from '../../../interfaces'
-import { BaseKey } from '../../common'
+import {
+  KeyValueStore,
+  ChainDB,
+  Transaction,
+  InclusionProof,
+} from '../../../interfaces'
+import {
+  BaseKey,
+  encode,
+  decode,
+  bnToUint256,
+  getTransactionRangeEnd,
+} from '../../common'
+
+// TODO: Where should this stuff sit? Should it be customizeable?
+const computeParent = (leftChild: string, rightChild: string): string => {
```

Comment:
@karlfloersch wasn't exactly sure how to handle this. We somehow need to provide the parent hashing function to this class.